### PR TITLE
Fixes #22704 Due date non english format

### PR DESCRIPTION
**Pull request by jdgrieco**
Get timestamp for non english due date format (Fixes #22704)

### CODE REVIEW

File: bug_change_status_page.php [hunk 1]
```diff
@@ -242,7 +242,7 @@
 		</th>
 		<td>
 			<input type="text" id="due_date" name="due_date" class="datetimepicker input-sm" size="16" maxlength="20"
-				data-picker-locale="<?php lang_get_current_datetime_locale() ?>"
+				data-picker-locale="<?php echo lang_get_current_datetime_locale() ?>"
```
Comment (vboctor) [inline bug_change_status_page.php]: This change has already been merged and it shouldn't be in this PR.

File: bug_update.php [hunk 2]
```diff
@@ -84,7 +84,8 @@
 	if( is_blank( $t_due_date ) ) {
 		$t_updated_bug->due_date = 1;
 	} else {
-		$t_updated_bug->due_date = strtotime( $t_due_date );
+	    $t_due_date = DateTime::createFromFormat( config_get( 'normal_date_format' ), $t_due_date );
```
Comment (vboctor) [inline bug_update.php]: Fix indentation.

File: core/bug_api.php [hunk 3]
```diff
@@ -314,7 +314,12 @@ public function __set( $p_name, $p_value ) {
 				break;
 			case 'due_date':
 				if( !is_numeric( $p_value ) ) {
-					$p_value = strtotime( $p_value );
+				    $p_value = DateTime::createFromFormat( config_get( 'normal_date_format' ), $p_value );
+				    if ( false === $p_value ) {
+				        $p_value = date_get_null();
+                    } else {
```
Comment (vboctor) [inline core/bug_api.php]: Fix indentation

File: bug_update.php (hunk 2 above)
Comment (vboctor) [inline bug_update.php]: Should you handle false value here?

File: core/bug_api.php [hunk 4]
```diff
@@ -314,7 +314,12 @@ public function __set( $p_name, $p_value ) {
 				break;
 			case 'due_date':
 				if( !is_numeric( $p_value ) ) {
-					$p_value = strtotime( $p_value );
+				    $p_value = DateTime::createFromFormat( config_get( 'normal_date_format' ), $p_value );
```
Comment (vboctor) [inline core/bug_api.php]: A couple of questions to consider:
- Should be using db_null() or triggering an error?
- Should this method or another method return date_get_null() when date is not valid?

File: core/bug_api.php (hunk 3 above)
Comment (jdgrieco) [inline core/bug_api.php]: @vboctor  Sorry about incorret indentation

**Comment by vboctor** [comment]
Commit messages should also include "Fixes #" followed by issue number (without space).

**Comment by jdgrieco** [comment]
@vboctor In that case I should not have put the PR in the issue description, right? Is the acceptable way to add this PR in a note?

**Comment by jdgrieco** [comment]
@syncguru As mentioned, I think this issue can exists in other date fields as well. Maybe in custom date fields! I not sure.

**Comment by cproensa** [comment]
The fixes seem that will be included in #1079
Let's keep it there to not interfere with @syncguru work

**Comment by cproensa** [comment]
This PR should be closed, as the original changes from #1032 have been reverted
The date picker will be reimplemented with the known issues in mind.

Anyway, thanks @jdgrieco for your submission

**Pull request closed without merging**